### ISSUE TITLE
automatically populate /tftpboot/

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#160]
**Reported by jenos on 7 Dec 2005 07:58 UTC**
```
Would be really nice if OSCAR took care of all manual
steps possible before the user runs "GO" (err,
install_cluster.. choose your name).  Super trivial
manual tasks like creating and populating /tftpboot/rpm
are exactly what OSCAR itself was chartered to do,
which kind of looks bad when oscar asks the user to do
that by him/herself as a very first step.

RPMS should be able to come from user's choice of
CD/DVD/Network...  with pre-coded servers available in
a list.  Auto failover from a down server wouldn't be
difficult to add for some extra polish to this feature.

As long as distro specific fingerprints are being
stored, it also wouldn't be a bad idea for this feature
to have some idea of how much space is required by the
files and to check for available disk space before
starting.

Seems as doable as any other distro specific feature
already in OSCAR, and it would be that much closer to
that ever tantalizing cluster installation fantasy of a
single step, big fat "GO" button.

```

**Comment by dikim33** [comment]
**Comment by valleegr on 16 Mar 2007 19:14 UTC**
Binary packages shipped with OSCAR are automatically copied into /tftpboot. Moreover it is possible to use an online repository. 
Therefore only scripts for the creation of a local mirror for the Linux distribution used is needed. That may be a tool (grab packages from an online repo for the creation of a local one).

**Comment by dikim33** [comment]
**Modified by jparpail on 20 Jul 2007 14:36 UTC**

**Comment by dikim33** [comment]
**Modified by valleegr on 30 Jun 2008 06:31 UTC**

**Comment by dikim33** [comment]
**Modified by valleegr on 12 Jan 2009 16:22 UTC**

**Comment by dikim33** [comment]
**Modified by valleegr on 12 Jan 2009 16:22 UTC**

**Comment by dikim33** [comment]
**Comment by valleegr on 15 Apr 2009 17:17 UTC**
(In [8399]) This check-in fixes #162, #514 - We can now mirror any repository and 
therefore fully populate /tftpboot if users want to do so.
